CI: reconsider fedora:rawhide

In https://github.com/OSGeo/gdal/pull/3813, we switched back again to fedora:34 because of failures of https://github.com/OSGeo/gdal/pull/3812/checks?check_run_id=2526882894
```
FAILED gcore/vsicrypt.py::test_vsicrypt_1 - ModuleNotFoundError: No module na...
FAILED gdrivers/mrf.py::test_mrf[jpeg/12bit_rose_extract.jpg-['COMPRESS=JPEG']]
FAILED gdrivers/netcdf.py::test_netcdf_51 - ModuleNotFoundError: No module na...
FAILED gdrivers/netcdf.py::test_netcdf_52 - ModuleNotFoundError: No module na...
FAILED gdrivers/netcdf.py::test_netcdf_62_cf_check - ModuleNotFoundError: No ...
FAILED gdrivers/nitf.py::test_nitf_43_jp2ecw - ModuleNotFoundError: No module...
FAILED gdrivers/vrtderived.py::test_vrtderived_10 - Failed: invalid checksum
FAILED gdrivers/vrtderived.py::test_vrtderived_12 - Failed: invalid checksum
ERROR gdrivers/nitf.py::test_nitf_41 - ModuleNotFoundError: No module named '...
ERROR gdrivers/nitf.py::test_nitf_42 - ModuleNotFoundError: No module named '...
ERROR gdrivers/nitf.py::test_nitf_online_14 - ModuleNotFoundError: No module ...
```
They seem related to failures when doing "import foo" where foo.py is one file in our test suite. Perhaps a change in python version

--- a/.github/workflows/fedora_rawhide.yml
+++ b/.github/workflows/fedora_rawhide.yml
@@ -28,4 +28,4 @@ jobs:
           restore-keys: ${{ runner.os }}-cache-fedora-rawhide-
 
       - name: Build
-        run: docker run --privileged=true -e CI -e WORK_DIR="$PWD" -v $PWD:$PWD -v /var/run/docker.sock:/var/run/docker.sock fedora:34 $PWD/.github/workflows/fedora_rawhide/start.sh
+        run: docker run --privileged=true -e CI -e WORK_DIR="$PWD" -v $PWD:$PWD -v /var/run/docker.sock:/var/run/docker.sock fedora:rawhide $PWD/.github/workflows/fedora_rawhide/start.sh

--- a/autotest/conftest.py
+++ b/autotest/conftest.py
@@ -44,10 +44,11 @@ def chdir_to_test_file(request):
     """
     old = os.getcwd()
 
-    os.chdir(os.path.dirname(request.module.__file__))
-    sys.path.insert(0, ".")
+    new_cwd = os.path.dirname(request.module.__file__)
+    os.chdir(new_cwd)
+    sys.path.insert(0, new_cwd)
     yield
-    if sys.path and sys.path[0] == ".":
+    if sys.path and sys.path[0] == new_cwd:
         sys.path.pop(0)
     os.chdir(old)
 

